clients/spark/core/src/main/scala/io/treeverse/gc/CommittedAddressLister.scala

```diff
@@ -0,0 +1,30 @@
+package io.treeverse.gc
+
+import org.apache.spark.sql.DataFrame
+import io.treeverse.clients.LakeFSContext
+import org.apache.spark.sql.SparkSession
+import io.treeverse.clients.LakeFSJobParams
+import org.apache.spark.sql.functions._
+
+trait CommittedAddressLister {
+  def listCommittedAddresses(spark: SparkSession, storageNamespace: String): DataFrame
+}
+
+class NaiveCommittedAddressLister extends CommittedAddressLister {
+  override def listCommittedAddresses(spark: SparkSession, storageNamespace: String): DataFrame = {
+    var normalizedStorageNamespace = storageNamespace
+    if (!normalizedStorageNamespace.endsWith("/")) {
+      normalizedStorageNamespace = "/"
+    }
+    val params =
+      LakeFSJobParams.forStorageNamespace(s"${normalizedStorageNamespace}", "uncommitted_gc")
+    var df = LakeFSContext.newDF(spark, params)
+    df = df
+      .select("address")
+      .withColumn("absolute_address", concat(lit(normalizedStorageNamespace), df("address")))
```

Comment:
Is it always a naive `concat`? I think some addresses might be relative..

Reply:
added a comment